Marlin/src/gcode/feature/pause/M600.cpp

```diff
@@ -71,42 +101,101 @@ void GcodeSuite::M600() {
   const float x_pos = parser.linearval('X', 0
     #ifdef PAUSE_PARK_X_POS
       + PAUSE_PARK_X_POS
+	#else
+      + current_position[X_AXIS]
     #endif
   );
   const float y_pos = parser.linearval('Y', 0
     #ifdef PAUSE_PARK_Y_POS
       + PAUSE_PARK_Y_POS
-    #endif
+	#else
+      + current_position[Y_AXIS]
+    #endif	
   );
-
+  
   // Unload filament
-  const float unload_length = parser.seen('U') ? parser.value_axis_units(E_AXIS) : 0
-    #if defined(FILAMENT_CHANGE_UNLOAD_LENGTH) && FILAMENT_CHANGE_UNLOAD_LENGTH > 0
-      - (FILAMENT_CHANGE_UNLOAD_LENGTH)
-    #endif
-  ;
-
+  #if ENABLED(ADVANCED_PAUSE_SPOOL_SWAP)
+    float unload_length = parser.seen('U') ? parser.value_axis_units(E_AXIS) : 0
+      #ifdef FILAMENT_CHANGE_UNLOAD_LENGTH
+        - FILAMENT_CHANGE_UNLOAD_LENGTH ;
+      if (swap_spool_enabled) unload_length = -ADVANCED_PAUSE_SPOOL_SWAP_UNLOAD;
+      #endif 
+  #else
```

Comment:
I'm sorry to write like this , i have no idea to make it better ,  i recalculate the value , already calculated
In pause , it's not a problem
If you have a better idea to avoid less computation for this